packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -901,7 +901,12 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
     }
 
     if (!this.state.focused) {
-      this.setState({ focused: true });
+      const state: Pick<IComboBoxState, 'focused'> | Pick<IComboBoxState, 'focused' | 'isOpen'> = { focused: true };
```

Comment:
I'd just use `Partial<IComboBoxState>`

Reply:
Me too, but that doesn't work.

Reply:
setState explicitly doesn't work with the Partial type, which is weird but that's the case :/

Reply:
Hmm, I think sometimes after using Partial I then have to do `this.setState(state as IState)`--which also doesn't seem like it should be necessary... Might be because Partial adds `| undefined` to each property, and if any of the state properties are required, the compiler doesn't like that.

Reply:
Looked up the signature for setState because I was curious... While this is technically correct, it does making correctly typing intermediate values (while avoiding casts) quite obnoxious.
`setState<K extends keyof S>(state: Pick<S, K> | S | null)`